sdk/core/azure-core/src/main/java/com/azure/android/core/internal/util/ExceptionUtils.java

```diff
@@ -22,6 +22,11 @@
     private ExceptionUtils() {
     }
 
+    /**
+     * A singleton IOException representing cancellation error.
+     */
+    public static final IOException CALL_CANCELLED_IO_EXCEPTION = new IOException("Canceled.");
```

Comment:
I preferred "Cancell****" for our "Type Names" mainly because the same is used in well-known [CancellationToken](https://docs.microsoft.com/en-us/dotnet/api/system.threading.cancellationtoken?view=netcore-3.1).

Our types are:

1. CancellationToken (public)
2. CancellationTokenImpl (internal)
3. ExceptionUtils.CALL_CANCELLED_IO_EXCEPTION (internal)

The reason we used `Canceled` in the "Exception message" of `IOException("Canceled.")` which we throw onCancel is to consistent with what OkHttp throws from its stack under the same condition, refer [this](https://github.com/square/okhttp/blob/4bfa33e09a4e472f8d8df402507d5bc0c4d3a416/okhttp/src/main/kotlin/okhttp3/internal/http/RetryAndFollowUpInterceptor.kt#L72), basically, we don't want customers to string match two different strings in case they have any logic based on the error message.

Reply:
I see, I agree that it makes sense to be consistent with OkHttp. I don't have a strong opinion on using either, however, I think we should have a default for every other case. I'll bring up with the team.